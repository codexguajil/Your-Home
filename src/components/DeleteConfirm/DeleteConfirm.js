import React, { Component } from 'react';
import { Modal, Button } from 'antd';
import { deleteProject, deleteRoom } from '../../thunks/fetchAllProjects';
import { connect } from 'react-redux';

export class DeleteConfirm extends Component {
  
  showConfirm(type, id, deleteProject, projectId, deleteRoom) {
    const confirm = Modal.confirm;
    confirm({
      title: 'Are you sure you want to delete this item and all contents within?',
      content: 'Click OK to confirm deletion.',
      onOk() {
        if(type === 'project') {
          deleteProject(id)
        }
        if(type === 'room') {
          console.log(id)
          deleteRoom(id, projectId)
        }
        if(type === 'material') {
          //deleteMaterial()//
        }
      },
      onCancel() {},
    });
  }

  render() {
    const {id, type, deleteProject, projectId, deleteRoom} = this.props
    return (
      <Button onClick={() => this.showConfirm(type, id, deleteProject, projectId, deleteRoom)} className='confirm-btn' type="link">
        <i className="fas fa-trash-alt"></i>
      </Button>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  deleteProject: (id) => dispatch(deleteProject(id)),
  deleteRoom: (id, projectId) => dispatch(deleteRoom(id, projectId))
})

export const mapStateToProps = state => ({
  projects: state.projects
})

export default connect(mapStateToProps, mapDispatchToProps)(DeleteConfirm);