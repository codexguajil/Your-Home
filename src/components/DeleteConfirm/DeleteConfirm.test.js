import React from 'react';
import { shallow } from 'enzyme';
import DeleteConfirm from './DeleteConfirm';

describe('DeleteConfirm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<DeleteConfirm />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should run show confirm when button is clicked', () => {
    // perhaps a simulation
    //
  });

  it('should confirm new data', () => {
    // prob check to see if correct functions are run depending on type
    //
  });

  //not sure if we need to test cancel?

});