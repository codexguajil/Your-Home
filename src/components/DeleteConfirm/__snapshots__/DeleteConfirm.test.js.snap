// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteConfirm should match snapshot 1`] = `
<Button
  block={false}
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[Function]}
  type="link"
>
  <i
    className="fas fa-trash-alt"
  />
</Button>
`;
